src/ast.hpp

```diff
@@ -228,37 +218,116 @@ namespace Sass {
   public:
     Vectorized(size_t s = 0) : hash_(0)
     { elements_.reserve(s); }
+    Vectorized(const std::vector<T> vec) : hash_(0)
+    { elements_ = vec; }
     virtual ~Vectorized() = 0;
     size_t length() const   { return elements_.size(); }
     bool empty() const      { return elements_.empty(); }
     void clear()            { return elements_.clear(); }
     T last() const          { return elements_.back(); }
     T first() const         { return elements_.front(); }
+
+    bool operator== (const Vectorized<T,U>& rhs) const {
+      // Abort early if sizes do not match
+      if (length() != rhs.length()) return false;
+      // Otherwise test each ast node in order
+      return std::equal(begin(), end(), rhs.begin(),
+        ObjEqualityFn<CssMediaQuery_Obj>);
```

Comment:
Why `CssMediaQuery_Obj`? Shouldn't it be `T`?

Reply:
Correct, thanks, will fix that